recipes/dgl/conda-build.patch

```diff
@@ -0,0 +1,222 @@
++else(EXTERNAL_DMLC_PATH)
++  target_include_directories(dgl PRIVATE "third_party/dmlc-core/include")
++  # For serialization
++  if (USE_HDFS)
++    option(DMLC_HDFS_SHARED "dgl has to build with dynamic hdfs library" ON)
++  endif()
++  add_subdirectory("third_party/dmlc-core")
++  list(APPEND DGL_LINKER_LIBS dmlc)
++  set(GOOGLE_TEST 0) # Turn off dmlc-core test
++endif(EXTERNAL_DMLC_PATH)
++
++if(EXTERNAL_PHMAP_PATH)
++  include_directories(SYSTEM ${EXTERNAL_PHMAP_PATH})
++else(EXTERNAL_PHMAP_PATH)
++  target_include_directories(dgl PRIVATE "third_party/phmap")
++endif(EXTERNAL_PHMAP_PATH)
++
++# this was removed as an explicit dependency in 1.1
++# target_include_directories(dgl PRIVATE "third_party/xbyak/")
++
++# this is always vendored 
+ target_include_directories(dgl PRIVATE "tensoradapter/include")
+-target_include_directories(dgl PRIVATE "third_party/nanoflann/include")
+-target_include_directories(dgl PRIVATE "third_party/libxsmm/include")
+-target_include_directories(dgl PRIVATE "third_party/pcg/include")
+ 
+-# For serialization
+-if (USE_HDFS)
+-  option(DMLC_HDFS_SHARED "dgl has to build with dynamic hdfs library" ON)
++if(EXTERNAL_NANOFLANN_PATH)
++  include_directories(SYSTEM ${EXTERNAL_NANOFLANN_PATH})
++else(EXTERNAL_NANOFLANN_PATH)
++  target_include_directories(dgl PRIVATE "third_party/nanoflann/include")
++endif(EXTERNAL_NANOFLANN_PATH)
++
++if (USE_LIBXSMM)
++  if (EXTERNAL_LIBXSMM_PATH)
++    include_directories(SYSTEM ${EXTERNAL_LIBXSMM_PATH})
++    if (NOT EXTERNAL_LIBXSMM_LIB_PATH)
++      message(FATAL_ERROR "EXTERNAL_LIBXSMM_LIB_PATH must be set if EXTERNAL_LIBXSMM_PATH is set")
++    endif()
++    find_package(xsmm
++      REQUIRED
++      HINTS ${EXTERNAL_LIBXSMM_LIB_PATH}
++    )
++    if(NOT xsmm_FOUND)
++        message(FATAL_ERROR "Failed to find LIBXSMM library")
++    endif()
++    list(APPEND DGL_LINKER_LIBS xsmm::xsmm)
++
++  else(EXTERNAL_LIBXSMM_PATH)
++    target_include_directories(dgl PRIVATE "third_party/libxsmm/include")
++    # Compile LIBXSMM
++    if(NOT MSVC)
++      set(LIBXSMM_COMPILER "${CMAKE_C_COMPILER}")
++      if (APPLE)
++        set(LIBXSMM_COMPILER "${CMAKE_C_COMPILER} -isysroot ${CMAKE_OSX_SYSROOT}")
++      endif()
++      if(REBUILD_LIBXSMM)
++        add_custom_target(libxsmm COMMAND make realclean COMMAND make -j ECFLAGS="-Wno-error=deprecated-declarations" BLAS=0 CC=${LIBXSMM_COMPILER}
++                          WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/libxsmm
++                          )
++      else(REBUILD_LIBXSMM)
++        add_custom_target(libxsmm COMMAND make -j ECFLAGS="-Wno-error=deprecated-declarations" BLAS=0 CC=${LIBXSMM_COMPILER}
++                          WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/third_party/libxsmm
++                          )
++      endif(REBUILD_LIBXSMM)
++      add_dependencies(dgl libxsmm)
++      list(APPEND DGL_LINKER_LIBS -L${CMAKE_SOURCE_DIR}/third_party/libxsmm/lib/ xsmm)
++    endif(NOT MSVC)
++  endif(EXTERNAL_LIBXSMM_PATH)
```

Comment:
@hmacdope regarding libxsmm I think it won't work as DGL uses newer version of libxssm which is not officialy released (latest is 1.17)
My idea is to build libxsmm from sources, but using header-only version - this probably would be compatible with conda-forge requirements

Please, don't commit this suggestion as it won't work with git patch 


To make it work changes in 'src/array/cpu/spmm_blocking_libxsmm.h' and 'src/runtime/config.cc' are required -
#include <libxsmm.h> ===> #include <libxsmm_source.h>
#include <libxsmm_cpuid.h> ===> #include <libxsmm_source.h>

To build in header-only mode replace all the code in USE_LIBXSMM condition with this single line

```suggestion
+ target_include_directories(dgl PRIVATE "third_party/libxsmm/include")
```

Reply:
Thanks so much again @bgawrych, I will make the required changes and combine them in one patch.